src/viam/sdk/robot/client.cpp

```diff
@@ -58,7 +58,9 @@ const std::string kStreamRemoved("Stream removed");
 
 RobotClient::~RobotClient() {
     if (should_close_channel_) {
-        this->close();
+        try {
```

Comment:
Alternatively, we could handle the exception in `close` and mark it `noexcept`. Or we could disable the check for escaping noexcept (which dtor's are, implicitly).